src/fsharp/TastOps.fs

```diff
@@ -5433,13 +5433,13 @@ let mapTargetsOfDecisionTree f tree = mapAccTipsOfDecisionTree (fun es i -> TDSu
 
 // Dead target elimination 
 let eliminateDeadTargetsFromMatch tree (targets:_[]) =
-    let used = accTargetsOfDecisionTree tree [] |> ListSet.setify (=) |> Array.ofList
-    if used.Length < targets.Length then
-        Array.sortInPlace used;
```

Comment:
What's happened to the sorting?

Reply:
reapplied sorting by using a F# set

Reply:
I don't get it - why not just keep the sortInPlace?  Converting to a binary tree and back just to sort a freshly allocated array is kind of wasteful.

Reply:
Yeah. This is a bit too much speculation here. Closing for now